Release/include/pplx/threadpool.h

```diff
@@ -156,6 +173,9 @@ class threadpool
     std::vector<pthread_t> m_threads;
     boost::asio::io_service m_service;
     boost::asio::io_service::work m_work;
+
+    std::mutex m_connection_pool_map_mutex;
+    std::map<std::string, std::shared_ptr<web::http::client::details::asio_connection_pool>> m_connection_pool_map;
```

Comment:
The map is get cleared when the program exits.  Although the connection pool will remain until the process exists, all actual connections in it have a default 30 seconds timer.  The timer is started after the connection being released, and the underlying socket is closed once the timer expires.

Using `std::weak_ptr` here will make the pool cleared right after the last `http_client` using it. If another `http_client` which is going to use the same pool is created after that, all connections have to be re-started.